src/main/java/com/example/BloodDonor/ServiceImplementation/AddressServiceImplementation.java

```diff
@@ -0,0 +1,119 @@
+package com.example.BloodDonor.ServiceImplementation;
+
+import com.example.BloodDonor.CustomException.DataNotFoundException;
+import com.example.BloodDonor.Entity.Address;
+import com.example.BloodDonor.Repository.AddressRepository;
+import com.example.BloodDonor.Service.AddressService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+
+@Service
+
+public class AddressServiceImplementation implements AddressService {
+    private static final Logger log= LoggerFactory.getLogger(AddressServiceImplementation.class);
+    @Autowired
+    private AddressRepository addressRepository;
+    @Override
+    public List<String> getAllCountries() {
+        log.info("get all service class is running");
+
+        List<String> list= addressRepository.findcountries();
+        return list;
+    }
+
+    @Override
+    public Address addAddressDetails(Address address) {
+        String City=address.getCity();
+        if(!addressRepository.existsByCity(City)){
+           return addressRepository.save(address);
+        }
+        return address;
+    }
+
+    @Override
+    public List<String> getAllStates(String country) throws DataNotFoundException{
+        if (!(addressRepository.existsByCountry(country))) {
+            log.warn("The country is not found "+country);
```

Comment:
follow the log pattern #
```suggestion
            log.warn("The country is not found "+country);
              log.warn("The country is not found {} ",country);
```